src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+    private Node<K, V>[] nodes;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        nodes = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (nodes.length * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            grow();
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = getIndexByKey(key);
+        if (nodes[index] == null) {
+            nodes[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> tempNode = nodes[index];
+        while (tempNode != null) {
+            if (Objects.equals(tempNode.key, key)) {
+                tempNode.value = value;
+                return;
+            }
+            if (tempNode.next == null) {
+                tempNode.next = newNode;
+                size++;
+                return;
+            }
+            tempNode = tempNode.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = nodes[getIndexByKey(key)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void grow() {
+        size = 0;
+        Node<K,V>[] oldNode = nodes;
+        nodes = new Node[nodes.length * 2];
+        for (Node<K, V> newNode : oldNode) {
+            while (newNode != null) {
+                put(newNode.key, newNode.value);
+                newNode = newNode.next;
+            }
+        }
+    }
+
+    private int getIndexByKey(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % nodes.length);
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
```

Comment:
don't forget about the access modifiers

Reply:
Can't make V final, because used it at 32 line, when assigning a value. And idea suggests changing the access modifier.

Reply:
i meant access modifier for the constructor make it private or public, but not a default